SSO Feature Set

zwwwenwei opened the pull request.

zwwwenwei committed:
Figma Implementation

zwwwenwei committed:
Figma Implementation

zwwwenwei committed:
Figma Implementation

zwwwenwei committed:
Figma Implementation

zwwwenwei committed:
Figma Implementation

zwwwenwei committed:
Figma Implementation

zwwwenwei committed:
Figma Implementation

zwwwenwei committed:
Update users_controller.rb

zwwwenwei committed:
Update authentication_controller.rb

zwwwenwei committed:
Update authentication_controller.rb

Josh-Piper reviewed (commented):
Can this Pull Request be put onto the auth-testing branch instead of main.

Josh-Piper commented on dream-big-api/app/controllers/users_controller.rb:
```diff
@@ -31,12 +30,13 @@ def update
       end
     end
-
+  
```
NIT: Whitespace not needed

Josh-Piper commented on dream-big-ui/src/app/Components/SolarSystem/section-dialog.component.html:
```diff
@@ -1,17 +1,247 @@
+<style>.mat-card{
+  background: linear-gradient(157.28deg, rgba(255, 255, 255, 0.4) 0%, #F5D5E0 100%);
+  width: 400px;
+height: 600px;
```
NIT: Whitespace

Josh-Piper commented on dream-big-ui/src/app/Components/SolarSystem/section-dialog.component.html:
```diff
@@ -1,17 +1,247 @@
+
+.wizard{
+  position: absolute;
+
```
NIT: Whitespace

Josh-Piper commented on dream-big-ui/src/app/Components/SolarSystem/section-dialog.component.html:
```diff
@@ -1,17 +1,247 @@
+   }
+
+   .name{
+    position: absolute;
```
NIT: Whitespace

Josh-Piper commented on dream-big-ui/src/app/Components/SolarSystem/section-dialog.component.ts:
```diff
@@ -10,6 +11,59 @@ export interface DialogData {
 	templateUrl: 'section-dialog.component.html',
 })
+
+
```
NIT: Whitespace

Josh-Piper commented on dream-big-ui/src/app/Components/SolarSystem/section-dialog.component.ts:
```diff
@@ -10,6 +11,59 @@ export interface DialogData {
+  color3 = "option";
+  color4 = "option";
+  color5 = "option";
+  color6 = "option";
```
What are these colours and options

Josh-Piper commented on dream-big-ui/src/app/Components/Wizard-Assessment/star-wizard/star-wizard.component.scss:
```diff
@@ -26,4 +26,17 @@
-}
\ No newline at end of file
+}
+
```
NIT: Whitespace

Josh-Piper commented on dream-big-ui/src/app/Components/Wizard-Assessment/star-wizard/star-wizard.component.ts:
```diff
@@ -14,10 +14,13 @@ export class StarWizardComponent implements OnInit {
     private stepsService: StepsService,
     private router: Router) { }
 
+
```
NIT: Whitespace

Josh-Piper reviewed (changes requested):
Can you add screenshots to the changes so it's obvious. Also try separating of concerns when creating a pull request. 

For example,

a pull request for the controller changes, a pull request for the Solar System changes etc.

zwwwenwei committed:
authentication_controller.rb

With a nested class called Authentication, I've made a new module called LoginLibrary. The SSO login as well as the standard login logic are both contained in the Authentication class.

Your AuthenticationController can use the library's methods rather than implementing the login logic yourself once the login system has been moved to the library.

zwwwenwei committed:
users_controller.rb

Installed and present in your project's Gemfile is the login_library gem. Depending on how your login library is implemented, modify the code as necessary.
